engine/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -0,0 +1,60 @@
+package cromwell.services
+
+import akka.actor.{Actor, Props}
+import com.typesafe.config.Config
+import cromwell.backend.BackendJobDescriptorKey
+import cromwell.core.WorkflowId
+import cromwell.services.MetadataServiceActor._
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+import org.joda.time.DateTime
+
+import scala.language.postfixOps
+
+object MetadataServiceActor {
+
+  final val MetadataServiceName = "MetadataService"
+
+  case class MetadataJobKey(callFqn: String, index: Option[Int], attempt: Int)
+  object MetadataJobKey {
+    def apply(backendJobKey: BackendJobDescriptorKey): MetadataJobKey = MetadataJobKey(backendJobKey.call.fullyQualifiedName, backendJobKey.index, backendJobKey.attempt)
+  }
+
+  case class MetadataKey(workflowId: WorkflowId, jobKey: Option[MetadataJobKey], key: String)
+  case class MetadataValue(value: String, timestamp: DateTime)
+  object MetadataValue {
+    def apply(value: String): MetadataValue = MetadataValue(value, DateTime.now)
+  }
+
+  case class MetadataEvent(key: MetadataKey, value: MetadataValue)
+
+  trait MetadataServiceMessage
+
+  /**
+    * Command Actions
+    */
+  trait MetadataServiceAction extends MetadataServiceMessage with ServiceRegistryMessage {
+    def serviceName = MetadataServiceName
+  }
+  case class MetadataPutAction(event: MetadataEvent) extends MetadataServiceAction
+  case class MetadataGetAction(workflowId: WorkflowId) extends MetadataServiceAction
```

Comment:
I think it's simple and better this way, by retrieving just using a WorkflowId (and letting the parsing/extracting logic to someone else)